File: part2/6_polynomial_regression/regression_template.R
# Regression template

dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Split data
library(caTools)
set.seed()
sample.split(y, SplitRatio = 2/3)

# Feature scaling


# Fitting the regression model to the dataset
# Create your regressor here


# Predicting a new result with the regressor
y_pred = predict(regressor, data.frame(Level = 6.5))

# Visualizing the regressor results
library(ggplot2)
ggplot() +
  geom_point(aes(x=dataset$Level,y=dataset$Salary), 
             color='red') +
  geom_line(aes(x=dataset$Level,y=predict(poly_reg, newdata = dataset)), 
            color='blue') +
  ggtitle('Level versus Salary') +
  xlab('Level') +
  ylab('Salary')

# Visualizing the regressor results - smoother curve
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
  geom_point(aes(x=dataset$Level,y=dataset$Salary), 
             color='red') +
  geom_line(aes(x=x_grid,y=predict(poly_reg, newdata = data.frame(Level = x_grid))), 
            color='blue') +
  ggtitle('Level versus Salary') +
  xlab('Level') +
  ylab('Salary')

File: part2/6_polynomial_regression/regression_template.py
# Regression template

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values

# Splitting the dataset into training and testing
'''from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)
'''

# Feature scaling


# Fitting the regression model to the dataset
# Create regressor here


# Predicting results
y_pred = regressor.predict(6.5)

# Visualizing the regression results
plt.scatter(X, y, color='red')
plt.plot(X, regressor.predict(X), color='blue')
plt.title('graph title')
plt.xlabel('x laebl')
plt.ylabel('y laebl')

# Visualizing the regression results (for higher resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X, y, color='red')
plt.plot(X_grid, regressor.predict(X_grid), color='blue')
plt.title('graph title')
plt.xlabel('x laebl')
plt.ylabel('y laebl')